Examples/BasketLosses/BasketLosses.cpp

```diff
@@ -18,7 +18,7 @@
 */
 
 #include <ql/qldefines.hpp>
-#ifdef BOOST_MSVC
+#if !defined(BOOST_ALL_NO_LIB) && defined(BOOST_MSVC)
```

Comment:
It's good that this showed we were defining this in `config.msvc.hpp`.  The things that get forgotten in the basement sometimes...
I'm kind of wondering if we want to have the Boost macro affect the linking of our library, though?  If we want a way to switch it off, I'd expect a `QL_` macro instead?

Reply:
That's up to you.  I dislike auto linking and avoid using it in cmake.  I had to add that extra clause you highlighted to ensure auto link was not activated.  You only need to link `boost::unit_test_framework` for the test suite and benchmark and possibly `boost::thread` based on the other QL flags.  The QuantLib library itself needs only Boost headers.  If QL clients specialize and instantiate template code that happens to require additional Boost libraries to operate, I would be surprised, but if so, it should be on them to add the links they need.  I believe auto link is something you should avoid not embrace.

Reply:
Philip, we are using MSVC in the following way:
- There is a solution file containing different projects, one of them is the `QuantLib.vcxproj` which is included as a git submodule of this repo. Here we are setting `QL_ENABLE_THREADSAFE_OBSERVER_PATTERN` in `userconfig.hpp` which auto links `Boost::Thread` for us.
- Now we also have within the solution a `QuantLibExtension.vcxproj` with our own functionality like classes derived from `Instrument`, `YieldTermStructure` etc. It only includes the QL and Boost headers and results in another static library.
- These two projects are linked together in a third one containing the current QL `swig` interface together with our extended functionality also exported via `swig` to `C#`. So here the two projects aka static libraries above are linked into a `dll`. These projects are included as VS project references so no extra setup happens here.
- Then there are more `C#` projects including the `swig` `C#` stuff, more `C#` functionality like database access, unit tests, `NuGet` packaging etc. using the QL and our extensions within `C#`.

For my understanding: will the auto linking still work for this setting for the third project building the `dll`?

Or what does

> If QL clients specialize and instantiate template code that happens to require additional Boost libraries to operate, I would be surprised, but if so, it should be on them to add the links they need.

mean under these circumstances?

> I believe auto link is something you should avoid not embrace.

Auto link does the trick for us when we update to a new version of QuantLib or Boost. We do **not** want to add the links we need.

Reply:
Auto linking works everywhere.  I just don't use it when building QL with my cmake code.   The workflow tests below for MSVC (the appveyor builds)  demonstrate that auto linking is working.